firedrake/mesh.py

```diff
@@ -451,6 +451,15 @@ def cell_closure(self):
         """
         pass
 
+    @property
+    @abc.abstractmethod
+    def entity_orientations(self):
+        """2D array of entity orientations
+
+        Each row contains orientations of the cell entities.
```

Comment:
Can you expand this docstring to explain what how the orientations are represented?

Reply:
Added more texts.